r/aws_elasticsearch_domain: export kibana endpoint

loivis opened the pull request:
Fixes #2802 

@fewknow, is this what you want, or rather export with `https://` scheme? The `endpoint` for elasticsearch domain itself is just a dns record, without any scheme.

fewknow commented:
@loivis looks like this will satisfy my use case.   I just wanted to be able to assign DNS to this in terraform so my end users can hit kibana.dev.corvestacloud.net.   

Thanks

Ninir commented on aws/resource_aws_elasticsearch_domain.go:
```diff
@@ -480,12 +484,14 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 		if err != nil {
 			return err
 		}
+		d.Set("kibana_endpoint", endpoints["vpc"].(string)+"/_plugin/kibana/")
```
Could we make use of a function that build this endpoint for us here and below?

loivis commented:
Hi @Ninir 

I added documentation for the attribute and extracted the function to populate kibana endpoint.

Regarding test, I just add simple check in acceptance test. Not sure if we need this check when setting the attribute. It is only basic string concatenation of `endpoint` attribute and `/_plugin/kibana/`

loivis commented:
```
⎇  make testacc TEST=./aws TESTARGS='-run=TestAccAWSElasticSearchDomain_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSElasticSearchDomain_basic -timeout 120m
=== RUN   TestAccAWSElasticSearchDomain_basic
--- PASS: TestAccAWSElasticSearchDomain_basic (870.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	870.636s
```

Ninir commented on website/docs/r/elasticsearch_domain.html.markdown:
```diff
@@ -109,6 +109,7 @@ The following attributes are exported:
 * `arn` - Amazon Resource Name (ARN) of the domain.
 * `domain_id` - Unique identifier for the domain.
 * `endpoint` - Domain-specific endpoint used to submit index, search, and data upload requests.
+* `kibana_endpoint` - Domain-specific endpoint for kibana without https scheme, the same as `endpoint`.
```
By "the same as `endpoint`", do you mean the same as in "without https"?
If so, I think this is misleading with the beginning of the sentence, what do you think? :)

loivis replied:
Exactly. Now removed those confusing words.

bflad commented:
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
